core/scheduler/metrics.go

```diff
@@ -78,7 +81,16 @@ func instrumentDuty(duty core.Duty, defSet core.DutyDefinitionSet) {
 	dutyCounter.WithLabelValues(duty.Type.String()).Add(float64(len(defSet)))
 }
 
-// instrumentValidator sets the validator balance.
-func instrumentValidator(pubkey core.PubKey, totalBal eth2p0.Gwei, status string) {
-	balanceGauge.WithLabelValues(string(pubkey), pubkey.String(), status).Set(float64(totalBal))
+// newMetricSubmitter returns a function that sets validator balance metric.
+func newMetricSubmitter() func(pubkey core.PubKey, totalBal eth2p0.Gwei, status string) {
+	prevStatus := make(map[core.PubKey]string)
+
+	return func(pubkey core.PubKey, totalBal eth2p0.Gwei, status string) {
+		balanceGauge.WithLabelValues(string(pubkey), pubkey.String(), status).Set(float64(totalBal))
+
+		if prev, ok := prevStatus[pubkey]; ok && prev != status { // Validator status changed
+			balanceGauge.WithLabelValues(string(pubkey), pubkey.String(), prev).Set(0)
```

Comment:
This will still result in the same problem, just with value==0ETH. 

The real problem is that we use the balance metric for two things. So instead of having two metrics that each do one thing, we have a dual purpose metric and this isn't correct.

The proper solution is to split the metric into two:
 - balance metic
 - constant value gauge with label=status.

Reply:
You're correct! Let me add in your suggestions